File: mockup/src/input-arg.js
// Input argument functions.


// Read string.
function getInputString(argArray)
{
	var arrayExists = Array.isArray(argArray);
	var argElement = null;
	var givenType = "";
	
	var stringRes = "";
	
	if (arrayExists === true && argArray.length > 2)
	{
		// Read command line argument.
		argElement = argArray[2];
		givenType = typeof argElement;
	}
	
	if (givenType === "string" && argElement.length > 0)
	{
		// Use value.
		stringRes = argElement;
	}
	
	return stringRes;
}


// Validate input string length.
function validateStringLength(inpLength)
{
	var maxLength = 100;
	var lengthValid = false;
	
	if (inpLength >= 0 && inpLength <= maxLength)
	{
		// Valid.
		lengthValid = true;
	}
	else if (inpLength > maxLength)
	{
		// Too long.
		throw new Error("Input string is too long.");
	}
	else
	{
		// Empty.
		throw new Error("Input string cannot be empty.");
	}
}


// Checks if input string is alphanumeric.
function validateStringCharacters(inpStr)
{
	var correctSyntax = /^[A-Z0-9]+$/gi;
	var matchFlag = inpStr.search(correctSyntax);
	
	if (matchFlag !== 0)
	{
		throw new Error("String must only contain alphanumeric characters.");
	}
}



module.exports =
{
	getString: getInputString,
	validateLength: validateStringLength,
	validateChars: validateStringCharacters
};

File: net-solution/PalindromePartitions/Tasks/InputArg.cs
using System;
using System.Text.RegularExpressions;

namespace PalindromePartitions.Tasks
{
	// Functions related to input text argument.
	public class InputArg
	{
		private static readonly Regex Syntax = new Regex("^[A-Z0-9]+$", RegexOptions.IgnoreCase);
		private const int MaxLength = 100;
		
		// Read text string.
		public static string ReadInputText(string[] argArray)
        {
            string inputRes;
			
			try
			{
				// Read argument.
				inputRes = argArray[0];
			}
			catch
			{
				// Argument does not exist.
				inputRes = "";
			}
			
			return inputRes;
		}
		
		// Validate string length and characters.
		public static bool ValidateInputText(ref string inpTxt)
		{
			bool correctSyntax = Syntax.IsMatch(inpTxt);
            bool validRes = false;
			
			if (inpTxt.Length > 0 && inpTxt.Length <= MaxLength && correctSyntax)
			{
				// Valid
				validRes = true;
			}
			else if (inpTxt.Length > 0 && inpTxt.Length <= MaxLength)
			{
				// Invalid characters.
                throw new Exception("Input string must only contain alphanumeric characters.");
			}
			else if (inpTxt.Length > MaxLength)
			{
				// Too long.
                throw new Exception("Input string can only have up to 100 characters.");
			}
			else
			{
				// Empty.
                throw new Exception("Input string is missing.");
			}
			
			return validRes;
		}
		
	}
}